### PR TITLE
external/iotivity: use snprintf instaed of sprintf

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- Replace sprintf to snprintf for safety

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3b3e92bc82a66942c59ebc54394d6d67ae08e938] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3b3e92bc82a66942c59ebc54394d6d67ae08e938] - Code Rule Check OK.

File: external/iotivity/iotivity_1.2-rel/resource/csdk/stack/src/oicgroup.c [hunk 1]
```diff
@@ -819,7 +819,7 @@ OCStackResult BuildStringFromActionSet(OCActionSet* actionset, char** desc)
     actionTypeStr = (char *)OICMalloc(1024);
     if(actionTypeStr != NULL)
     {
-        sprintf(actionTypeStr, "%ld %u", actionset->timesteps, actionset->type);
+        snprintf(actionTypeStr, sizeof(actionTypeStr), "%ld %u", actionset->timesteps, actionset->type);
```
Comment (zhouxinhe) [inline external/iotivity/iotivity_1.2-rel/resource/csdk/stack/src/oicgroup.c]: `sizeof(actionTypeStr)`==`sizeof(char *)`, I think you mean this:
 ```suggestion
        snprintf(actionTypeStr, 1024, "%ld %u", actionset->timesteps, actionset->type);
```

**Comment by SeonghoByeon** [comment]
> The second parameter of the callexpr is wrong. The code calls sizeof() on a malloced pointer type, which always returns the size of the pointer, not the size of an allocated buffer. For details, see https://cwe.mitre.org/data/definitions/467.html
> 
> This parameter must be related to the magic number '1024'

Thanks for the comment.
Actually, I prepared a new PR for this issue.
Please refer to https://github.com/Samsung/TizenRT/pull/3715.

**Pull request merged**